hardhat.config.ts

```diff
@@ -0,0 +1,129 @@
+// Hardhat plugins
+// Note: Order probably matters!
+import "@nomicfoundation/hardhat-chai-matchers";
+import "@nomicfoundation/hardhat-network-helpers";
+import "@nomiclabs/hardhat-ethers";
+import "hardhat-abi-exporter";
+import "hardhat-diamond-abi";
+// Must be registered after hardhat-diamond-abi
+import "@typechain/hardhat";
+import "hardhat-contract-sizer";
+import "hardhat-settings";
+import "@solidstate/hardhat-4byte-uploader";
+
+// Other
+import type { HardhatUserConfig } from "hardhat/config";
+import * as mapWorkspaces from "@npmcli/map-workspaces";
+
+const packages = mapWorkspaces.virtual({
+  cwd: __dirname,
+  pkg: require("./package.json"),
+  lockfile: require("./package-lock.json"),
+});
+
+const config: HardhatUserConfig = {
+  defaultNetwork: "hardhat",
+  /**
+   * This configures which networks hardhat will communicate with
+   */
+  networks: {
+    localhost: {
+      url: "http://localhost:8545/",
+      accounts: {
+        mnemonic:
+          "change typical hire slam amateur loan grid fix drama electric seed label",
```

Comment:
thoughts on moving these to variables in a `.env`/`.env.example` file?

I see exgrasia (so presumably some old version of DF also) used to do that https://github.com/nalinbhardwaj/exgrasia/blob/master/eth/hardhat.config.ts#L20

Reply:
Yep! DF still does this: https://github.com/darkforest-eth/eth/blob/master/hardhat.config.ts#L35-L37 and even determines which networks to add based on the deployer being available: https://github.com/darkforest-eth/eth/blob/master/hardhat.config.ts#L91-L95

One thing that I'd like to solve is supporting a private key or mnemonic before we add support for that. Currently this only has local networks, so I figured I could defer those improvements until later.